app/services/duration_parser.rb

```diff
@@ -0,0 +1,39 @@
+# Parses duration strings ("1d", 2w", "3m", "4y" into ActiveSupport::Durations)
+class DurationParser
+  attr_reader :value
+
+  # @param value [String, nil]
+  def initialize(value)
+    @value = value
+  end
+
+  # @return [ActiveSupport::Duration, nil]
+  def parse
+    return if value.blank?
+
+    match = value.match(/^(?<number>\d+)(?<duration>\D)$/)
```

Comment:
I guess the switch fallthrough would have the same effect of generating a `nil` value, but... should we match on the specific duration characters we support?

```suggestion
    match = value.match(/^(?<number>\d+)(?<duration>[dwmy])$/)
```

Reply:
My thinking was the switch fallthrough would be the easiest? If we ever add a new character we only need to add it on one place, also allows like multicharacter signifiers if we ever wanted to allow like `hr` and `wk` or something

Reply:
> My thinking was the switch fallthrough would be the easiest? If we ever add a new character we only need to add it on one place, also allows like multicharacter signifiers if we ever wanted to allow like `hr` and `wk` or something

Seems fine to me. And I just now noticed you have test coverage for this with the `123x` example, so should be protected against accidental regressions. On that point, I guess my primary concern is not that the switch fallthrough doesn't work, just that it might not be as obvious to the next person that we're relying on it.